### PR TITLE
fix selection_rate for y_pred of length 0 or 1

### PR DESCRIPTION
**Pull request by romanlutz**
Currently the squeeze call causes an issue for `y_pred` of length 1. Additionally, I've added a check to ensure it's not empty either.

Signed-off-by: Roman Lutz <rolutz@microsoft.com>

### CODE REVIEW

File: fairlearn/metrics/_selection_rate.py [hunk 1]
```diff
@@ -9,7 +9,18 @@ def selection_rate(y_true, y_pred, *, pos_label=1, sample_weight=None):
 
     The argument `pos_label` specifies the 'good' outcome.
     """
-    selected = (np.squeeze(np.asarray(y_pred)) == pos_label)
+    y_pred_arr = np.asarray(y_pred)
+
+    if y_pred_arr.shape == (0,):
```
Comment (riedgar-ms) [inline fairlearn/metrics/_selection_rate.py:14]: Why not use:
https://github.com/fairlearn/fairlearn/blob/4f8fd73466bf34e4fb12647c767f1144fa14d060/fairlearn/metrics/_input_manipulations.py#L9

Comment (romanlutz) [reply fairlearn/metrics/_selection_rate.py:14]: I suppose that works as well!

File: fairlearn/metrics/_selection_rate.py [hunk 2]
```diff
@@ -3,13 +3,18 @@
...
+
 
 def selection_rate(y_true, y_pred, *, pos_label=1, sample_weight=None):
     """Calculate the fraction of predicted labels matching the 'good' outcome.
 
     The argument `pos_label` specifies the 'good' outcome.
     """
-    selected = (np.squeeze(np.asarray(y_pred)) == pos_label)
+    if len(y_pred) == 0:
+        raise ValueError("Empty y_pred passed to selection_rate function.")
```
Comment (riedgar-ms) [inline fairlearn/metrics/_selection_rate.py:15]: Can you add a couple of tests, to avoid regressions?

File: fairlearn/widget/js/yarn.lock [hunk 3]
```diff
@@ -1255,10 +1255,10 @@ extglob@^2.0.4:
     snapdragon "^0.8.1"
     to-regex "^3.0.1"
 
-fairlearn-dashboard@0.0.21:
-  version "0.0.21"
-  resolved "https://registry.yarnpkg.com/fairlearn-dashboard/-/fairlearn-dashboard-0.0.21.tgz#644eff7bf408317cbaf8c9e197888ac61a817ac0"
-  integrity sha512-vxBnnzmFDeE978VsY/JvwtYSHIfW105LVBLl9KFg8sfpYk4EIHDPdAV/djp2XEarDaHQ4gJUycp1zmpHMNowjg==
+fairlearn-dashboard@0.0.22:
```
Comment (riedgar-ms) [inline fairlearn/widget/js/yarn.lock:1258]: Ermmm....?

**Review by riedgar-ms — CHANGES_REQUESTED** [review]
Can you kick out all the JS changes, please?

File: fairlearn/widget/js/yarn.lock (hunk 3 above)
Comment (romanlutz) [reply fairlearn/widget/js/yarn.lock]: Good point. Must have been lingering around from a local build for something else... removed this via rebase -i

**Review by riedgar-ms — APPROVED** [review]

**Pull request merged**